### PR TITLE
test: remove Enzyme, move existing tests to React Testing Library

### PR DESCRIPTION
**Pull request by lemald**
#### Summary of changes
**Asana Ticket:** [🏹 Transition existing Arrow tests from Enzyme to React Testing Library](https://app.asana.com/0/584764604969369/1170122874093785/f)

This is mostly pretty rote replacement of Enzyme with React Testing Library. The one bit I'm still not super happy with is the test `selecting a day of the week enables updating time range`. It used to do a lot of asserting based on the props of the component to make sure it was doing the right thing with the day of week time ranges. However, React Testing Library is a lot more opinionated about not really letting you pry into the contents of React components and instead preferring you to strictly test the front-end. I'm open to suggestions on how to re-do that test in a nicer way if anyone has any.

#### Reviewer Checklist
- [ ] Meets ticket's acceptance criteria
- [ ] Any new or changed functions have typespecs
- [ ] Tests were added for any new functionality (don't just rely on Codecov)
- [ ] This branch was deployed to the staging environment and is currently running with no unexpected increase in warnings, and no errors or crashes.

### CODE REVIEW

File: assets/tests/disruptions/disruptionTimePicker.test.tsx [hunk 1]
```diff
@@ -28,334 +28,347 @@ const DisruptionTimePickerWithProps = ({}): JSX.Element => {
...
+
+    const mondayEndPeriodSelect = container.querySelector(
+      "select#time-of-day-end-period-0"
+    )
+    if (!mondayEndPeriodSelect) {
+      throw new Error("Monday end period select not found")
+    }
+    fireEvent.change(mondayEndPeriodSelect, { target: { value: "PM" } })
+
+    fireEvent.click(mondayEndTypeCheck)
```
Comment (fjlanasa) [inline assets/tests/disruptions/disruptionTimePicker.test.tsx:108]: Are these last two click events necessary? Seems like there should be an assertion afterwards, or else we can remove.

Comment (lemald) [reply assets/tests/disruptions/disruptionTimePicker.test.tsx:108]: Without these last couple of clicks there's a line or two that goes uncovered. This is the test that my note in the PR description is referring to, previously there were some more assertions in the test but there isn't really a good way to directly test the value of the props anymore, at least that I'm aware of.

Comment (fjlanasa) [reply assets/tests/disruptions/disruptionTimePicker.test.tsx:108]: Could you just assert on the state of the input? Like I assume clicking `mondayEndTypeCheck` or `mondayCheck` at that point disables and/or changes the value of some inputs.

Comment (lemald) [reply assets/tests/disruptions/disruptionTimePicker.test.tsx:108]: Yeah, it should disable the dropdowns. That's probably the most sensible thing to assert.

**Pull request merged**